tests/e2e/tests/checkout/card-failures.spec.js

```diff
@@ -0,0 +1,47 @@
+const { test, expect } = require( '@playwright/test' );
+import config from 'config';
+const {
+	emptyCart,
+	setupProductCheckout,
+	setupCheckout,
+	fillCardDetails,
+} = require( '../../utils/payments' );
+
+test.beforeEach( async ( { page } ) => {
+	await emptyCart( page );
+	await setupProductCheckout( page );
+	await setupCheckout( page, config.get( 'addresses.customer.billing' ) );
+} );
+
+const testFunctionForCard = async ( page, cardKey ) => {
```

Comment:
Nit: Does the "Function" in the function name stand for "functionality", or is it a redundancy thing, like naming a variable A with `variableA` ? 😀 

In both cases, I'd suggest something like `testCardFailure`, but feel free to ignore.

Reply:
Good catch! It was redundancy like using `variableA` for a variable :laughing: 
The name was changed to `async testCard( page, cardKey )`.